### PR TITLE
React controlled Dropdowns

### LINKED ISSUE #151: Use react-popper for Dropdowns

### PR DESCRIPTION
**Pull request by jonthomp**
First, this rearranges the Dropdown internals to hopefully make them a bit clearer, then it adds the various bits and pieces to make our Dropdowns React powered rather than bootstrap/jQuery.

For dropdowns created via the `Dropdown` props, this uses [react-popper](https://github.com/FezVrasta/react-popper) for positioning of `DropdownMenu`s near their `DropdownTrigger` and state built into the `Dropdown` for handling showing/hiding of the menu.

Resolves #151

### CODE REVIEW

File: src/components/Button/Button.react.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ type PropsForAll = {|
   +isDropdownToggle?: boolean,
   +to?: string,
   +isOption?: boolean,
+  +rootRef?: Function,
```
Comment (AaronCoplan) [inline src/components/Button/Button.react.js]: Let's type this a little more strictly if possible i.e. `(a: string, b: bool) => void`

File: src/components/Dropdown/DropdownMenu.react.js [hunk 2]
```diff
@@ -3,14 +3,30 @@
...
-   * Display an arrow poiting towards the trigger
+   * Display an arrow pointing towards the trigger
    */
   +arrow?: boolean,
+  /**
+   * The position of the arrow pointing towards the trigger
+   */
+  +arrowPosition?: "left" | "right",
+  +style?: Style & StyleOffsets & StylePosition,
+  +rootRef?: Function,
```
Comment (AaronCoplan) [inline src/components/Dropdown/DropdownMenu.react.js]: Same thing with stricter function typing.

File: src/components/Dropdown/DropdownTrigger.react.js [hunk 3]
```diff
@@ -35,6 +35,11 @@ type Props = {|
    * Is this trigger beind used as a Card.Header option
    */
   +isOption?: boolean,
+  /**
+   * Handle the onClick of this trigger
+   */
+  +onClick?: () => void,
+  +rootRef?: Function,
```
Comment (AaronCoplan) [inline src/components/Dropdown/DropdownTrigger.react.js]: Same thing here

**Comment by jonthomp** [comment]
@AaronCoplan have made those amends.

On react-popper, does quite a lot for us under the hood, it handles the positioning of the Dropdown menu, takes into account scrolling, viewport, containing elements etc and re-adjusts the positioning to make sure it's where we want it and visible.

**Comment by AaronCoplan** [comment]
Awesome! Good stuff.  I was just wondering and didn't have any context - thats great!

**Pull request merged**